lib/application/pages/home/collections/update/update_collection_memos.dart

```diff
@@ -1,18 +1,168 @@
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final vm = ref.read(updateCollectionMemosVM.notifier);
+    final state = ref.watch(updateCollectionMemosVM);
+    final currentPageIndex = useState(0);
+
+    final controller = usePageController(viewportFraction: 0.95);
+    final parentVM = ref.watch(updateCollectionVM.notifier);
+    ref.listen<UpdateMemosState>(updateCollectionMemosVM, (_, state) => parentVM.updateMemos(memos: state.memos));
+
+    useEffect(() {
+      void onPageUpdate() => currentPageIndex.value = controller.page!.toInt();
+
+      controller.addListener(onPageUpdate);
+      return () => controller.removeListener(onPageUpdate);
+    }, []);
+
+    // Uses `PageView.custom` to support pages reordering.
+    final pagesView = PageView.custom(
+      controller: controller,
+      childrenDelegate: SliverChildBuilderDelegate(
+        (context, index) {
+          if (index == state.memos.length) {
+            return _CreateMemoEmptyState(onTap: vm.createEmptyMemo);
+          }
+
+          final metadata = state.memos[index];
+
+          return _MemoPage(
+            key: ValueKey(metadata),
+            pageIndex: index,
+            metadata: metadata,
+            onRemove: state.memos.length > 1 ? () => vm.removeMemoAtIndex(index) : null,
+          ).withOnlyPadding(context, right: Spacing.xSmall);
+        },
+        // Adds +1 to include creation empty state.
+        childCount: state.memos.length + 1,
+        // Returns the memo index in case of reordering.
+        findChildIndexCallback: (key) {
+          final valueKey = key as ValueKey<MemoUpdateMetadata>;
+          return state.memos.indexOf(valueKey.value);
+        },
+      ),
+    );
+
+    void onPreviousTapped() =>
+        controller.previousPage(duration: anims.pageControllerAnimationDuration, curve: anims.defaultAnimationCurve);
+    void onNextTapped() =>
+        controller.nextPage(duration: anims.pageControllerAnimationDuration, curve: anims.defaultAnimationCurve);
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        Expanded(child: pagesView),
+        context.verticalBox(Spacing.large),
+        _NavigationIndicator(
+          // Adds +1 to transform index into page
+          currentPage: currentPageIndex.value + 1,
+          // Adds +1 to include the creation CTA
+          pagesAmount: state.memos.length + 1,
+          onLeftTapped: currentPageIndex.value > 0 ? onPreviousTapped : null,
+          onRightTapped: currentPageIndex.value < state.memos.length ? onNextTapped : null,
+        ),
+      ],
+    ).withSymmetricalPadding(context, vertical: Spacing.medium);
+  }
+}
+
+/// Presents editable `Memo` [metadata] wrapped around [UpdateMemoTerminal].
+class _MemoPage extends HookConsumerWidget {
+  const _MemoPage({required this.pageIndex, required this.metadata, this.onRemove, Key? key}) : super(key: key);
+
+  final int pageIndex;
```

Comment:
Could add docs here on what this index means

Reply:
Should the child know its position in the parent? Looks to me like it would be easier for the parent, that controls the list, to know where it's inserted and what it should do with a given child event.

Reply:
I've moved the responsibility of warning the VM about the changes to the parent, but we still have to have this index here because we visually show to the user what question is he seeing:

<details>
<summary>Screenshot</summary>

![CleanShot 2021-11-29 at 11 17 55@2x](https://user-images.githubusercontent.com/11745745/143884803-67c10e52-5b82-4181-87c7-673a49bb31a6.png)

</details>

But now it only serve as a visual indicator, and it's not used anymore to update the VM